go/types/set_iterator.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package types
+
+import "github.com/attic-labs/noms/go/d"
+
+type SetIterator interface {
+	// Next returns nil, when no objects remain
+	Next() Value
+
+	// AdvanceTo(v) returns:
```

Comment:
Another independent comment here: looking at the tests, the behavior here does seem reasonable, though the method might be better named similar to `NextFrom`. `AdvanceTo` sounds like it does just take the iterator to a position such that the next call to `Next` returns that value.